src/kits/tracker/ContainerWindow.cpp

```diff
@@ -3309,22 +3270,26 @@ void
 BContainerWindow::_AddFolderIcon()
 {
 	if (fMenuBar == NULL)
-		return;
+		return; // We don't want to add the icon if there's no menubar
+	
 	float iconSize = fMenuBar->Bounds().Height() - 2;
 	if (iconSize < 16)
 		iconSize = 16;
-	float iconPosY = 1 + (fMenuBar->Bounds().Height() - 2
-		- iconSize) / 2;
-	BView* icon = new(std::nothrow) DraggableContainerIcon(
-		BRect(Bounds().Width() - 4 - iconSize + 1, iconPosY,
-			Bounds().Width() - 4, iconPosY + iconSize - 1), "ThisContainer",
-		B_FOLLOW_RIGHT);
-	if (icon != NULL)
-		fMenuBar->AddChild(icon);
+	
+	menu_info info;
+	get_menu_info(&info);
```

Comment:
That ugly code could be removed when using the B_MENU_BACKGROUND_COLOR code from above.